tower-balance/src/lib.rs

```diff
@@ -106,6 +106,26 @@ where
         }
     }
 
+    /// Returns true iff there are ready services.
+    pub fn is_ready(&self) -> bool {
+        !self.ready.is_empty()
```

Comment:
Looking through the `impl` again, it seems like the check also needs to factor in `dispatched_ready_index`.

Specifically, if `dispatched_ready_index` is `Some`, it is possible for an entry in `ready` to not *actually* be empty.

I would also add a comment stating that even if `is_ready` returns `true`, you still have to call `poll_ready` (otherwise, panic).

Reply:
yeah... these calls only make sense after poll_ready is called. we don't _need_ to expose these publicly, even, so perhaps better to remove?  i'll clarify the comments at least, though.